### PR TITLE
feat: add optional third angle to MS gate

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Add optional third angle to IonQ native MS gate.

On the service, fails due to lack of support for third angle: `botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the CreateQuantumTask operation: [line 4] invalid number of argument(s). Expected: 2, Got: 3`

*Testing done:*

Unit testing and manual testing (fails today, will repeat before merging)

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/circuits/angled_gate.py [hunk 1]
```diff
@@ -259,27 +386,27 @@ def angled_ascii_characters(gate: str, angle: Union[FreeParameterExpression, flo
...
     """
     return (
         f"{gate}("
-        f'{angle_1:{".2f" if isinstance(angle_1, (float, Float)) else ""}}, '
-        f'{angle_2:{".2f" if isinstance(angle_2, (float, Float)) else ""}})'
+        + ", ".join(
+            f'{angle:{".2f" if isinstance(angle, (float, Float)) else ""}}' for angle in angles
+        )
+        + ")"
     )
```
[the remark this answers is not in the document]
Comment (ajberdy) [reply src/braket/circuits/angled_gate.py:410]: Sure, that works too

File: src/braket/circuits/angled_gate.py [hunk 2]
```diff
@@ -232,6 +232,133 @@ def __repr__(self):
...
+                target qubit on the second index, the ASCII symbols should have `["C", "X"]` to
+                correlate a symbol with that index.
+
+        Raises:
+            ValueError: If `qubit_count` is less than 1, `ascii_symbols` are `None`, or
+                `ascii_symbols` length != `qubit_count`, or `angle_1` or `angle_2` or `angle_3`
+                 is `None`
+        """
+        super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
+        if angle_1 is None or angle_2 is None or angle_3 is None:
```
[the remark this answers is not in the document]
Comment (ajberdy) [reply src/braket/circuits/angled_gate.py:270]: This is the standard way of handling angle parameters for all gates. If there are edge cases missed, we should make the update across the board in a separate PR

File: src/braket/circuits/angled_gate.py [hunk 3]
```diff
@@ -259,27 +386,27 @@ def angled_ascii_characters(gate: str, angle: Union[FreeParameterExpression, flo
     return f'{gate}({angle:{".2f" if isinstance(angle, (float, Float)) else ""}})'
 
 
-def _double_angled_ascii_characters(
+def _multi_angled_ascii_characters(
     gate: str,
-    angle_1: Union[FreeParameterExpression, float],
-    angle_2: Union[FreeParameterExpression, float],
+    *angles: List[Union[FreeParameterExpression, float]],
```
[the remark this answers is not in the document]
Comment (ajberdy) [reply src/braket/circuits/angled_gate.py:391]: Just checked and seems like `*args` parameters should be annotated with the type of a single arg instance. I'll update to remove the List type

File: src/braket/circuits/angled_gate.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajberdy) [reply src/braket/circuits/angled_gate.py:410]: Actually `.2f` is consistent with existing code for single angled gates too. This change could have unforeseen consequences, so I'd prefer to do that in a separate PR and make the related updates together.

Thought experiment, an ISV could be parsing gate strings expecting 2 decimals and this breaks their code. Unlikely, but enough of a reason to separate the change imo

**Review by mbeach-aws — APPROVED** [review]

**Review by speller26 — APPROVED** [review]

**Pull request merged**